cmd/soci/commands/ztoc/list.go

```diff
@@ -30,25 +34,85 @@ var listCommand = cli.Command{
 	Description: "list ztocs",
 	Flags: []cli.Flag{
 		cli.StringFlag{
-			Name:  "digest",
+			Name:  "ztoc-digest",
 			Usage: "filter ztocs by digest",
 		},
+		cli.StringFlag{
+			Name:  "image-ref",
+			Usage: "filter ztocs to those that are associated with a specific image",
+		},
 	},
 	Action: func(cliContext *cli.Context) error {
 		db, err := soci.NewDB()
 		if err != nil {
 			return err
 		}
-		digest := cliContext.String("digest")
+		ztocDgst := cliContext.String("ztoc-digest")
+		imgRef := cliContext.String("image-ref")
 
 		var artifacts []*soci.ArtifactEntry
-		db.Walk(func(ae *soci.ArtifactEntry) error {
-			if ae.Type == soci.ArtifactEntryTypeLayer &&
-				(digest == "" || ae.Digest == digest) {
-				artifacts = append(artifacts, ae)
+		if imgRef == "" {
+			db.Walk(func(ae *soci.ArtifactEntry) error {
+				if ae.Type == soci.ArtifactEntryTypeLayer && (ztocDgst == "" || ae.Digest == ztocDgst) {
+					artifacts = append(artifacts, ae)
+				}
+				return nil
+			})
+		} else {
+			client, ctx, cancel, err := commands.NewClient(cliContext)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			is := client.ImageService()
+			img, err := is.Get(ctx, imgRef)
+			if err != nil {
+				return err
+			}
+			platform, err := images.Platforms(ctx, client.ContentStore(), img.Target)
+			if err != nil {
+				return err
 			}
-			return nil
-		})
+			var layers []ocispec.Descriptor
+			for _, p := range platform {
+				manifest, err := images.Manifest(ctx, client.ContentStore(), img.Target, platforms.OnlyStrict(p))
+				if err != nil {
+					// print a warning message if a manifest can't be resolved
+					// continue looking for manifests of other platforms
+					fmt.Printf("no image manifest for platform %s/%s. err: %v\n", p.Architecture, p.OS, err)
```

Comment:
Possibly for a follow up, but can we hide this behind a `--verbose` flag or something? I would expect that in most cases a user is working with a single platform and this just adds unnecessary clutter. It could be useful for debugging if you created indices for all platforms and then deleted the image, for example. The ztocs would still be there, but the filtering wouldn't work.

Reply:
Honestly we can just not print anything in this case, and only add the layers from the available manifests. The purpose of ``soci ztoc list`` command is to filter out all _available_ ztocs. What platforms the image is not on is not really its concern.

Reply:
Added ``--verbose`` flag